### PR TITLE
Remove unused client data-fetching code for`/deposits/overview`

### LINKED ISSUE #8047: Remove unused React hook `useDepositsOverview` and related data fetching code
### Describe the bug

We have an unused React hook `useDepositsOverview` that we no longer use (since WooPayments [2.5](https://github.com/Automattic/woocommerce-payments/releases/tag/2.5.0)). We should remove this hook and related `/deposits/overview` data-fetching code to improve the quality of our codebase and to ensure we're not unnecessarily maintaining dead code.

https://github.com/Automattic/woocommerce-payments/blob/a7904e5215b7be90203defa7e6b928c772aa0a43/client/data/deposits/hooks.ts#L82-L99

This hook has been superseded by the [`useAllDepositsOverviews` hook.](https://github.com/Automattic/woocommerce-payments/blob/a7904e5215b7e90203defa7e6b928c772aa0a43/client/data/deposits/hooks.ts#L101)

### See also
* server issue for removal of this endpoint from server: https://github.com/Automattic/woocommerce-payments-server/issues/4489
* client issue for removal of this endpoint from client: https://github.com/Automattic/woocommerce-payments/issues/8179

### PR DESCRIPTION
**Pull request by Jinksi**
Fixes #8047

#### Changes proposed in this Pull Request

This PR removed unused data-fetching code related to the now redundant `/deposits/overview` server endpoint:
* any `wp.data` code related to fetching `/deposits/overview`
* the `useDepositsOverview` react hook

Out of scope:
* the `/deposits/overview` REST API endpoint (see #8047)

The `/deposits/overview-all` endpoint has been used instead since WooPayments [2.5](https://github.com/Automattic/woocommerce-payments/releases/tag/2.5.0).

#### Testing instructions

* This code was unused – ensure GH checks pass

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests **Redundant tests have been removed**
- [x] Tested on mobile **N/A**

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `8178` or branch name `fix/8047-remove-unused-deposits-overview-code` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/8047-remove-unused-deposits-overview-code) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: debfb7ca634451f3c87a6e8b0902647d5f8c5983
- Build time: 2024-02-13 22:34:34 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/data/deposits/test/reducer.js [hunk 1]
```diff
@@ -130,40 +125,6 @@ describe( 'Deposits reducer tests', () => {
...
-		);
-
-		expect( reduced ).toStrictEqual( {
-			...mockEmptyState,
-			overview: {
-				error: { code: 'error' },
-			},
-		} );
-	} );
-
```
Comment (haszari) [inline client/data/deposits/test/reducer.js:166]: I was not aware we had tests for reducers! I wonder how many issues these have caught.

**Review by haszari — APPROVED** [review]
Code removal looks good – always a bit confronting seeing how much code is needed to fetch & wrangle data from an API (redux/resolvers style)!

I did some quick testing of deposits screens on my test store and saw no issues.

Note my test store doesn't have any successful scheduled deposits (not sure why, unrelated), so please ensure you've tested on a store with typical deposits history data. 

Approving, 100% confident this is no risk and great to tidy up!

**Comment by Jinksi** [comment]
> please ensure you've tested on a store with typical deposits history data.

✅ tested against stores in a non-local environment with typical deposit history data.

**Pull request merged**